components/subscription-base-infra/main.tf

```diff
@@ -36,3 +36,10 @@ resource "azurerm_storage_container" "container" {
   storage_account_name  = azurerm_storage_account.storage_account.name
   container_access_type = "private"
 }
+
+resource "azurerm_key_vault_secret" "env_subscription_id" {
+  count        = var.env == "ptl" ? 1 : 0
+  name         = "${var.env}-subscription-id"
+  value        = var.prod_subscription_id
```

Comment:
this should come from the subscription you're current running in using a data source